src/types.rs

```diff
@@ -287,23 +292,25 @@ impl PartialOrd for Mode {
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         match (*self, *other) {
             (a, b) if a == b => Some(Ordering::Equal),
-            (Mode::Numeric, Mode::Alphanumeric) | (_, Mode::Byte) => Some(Ordering::Less),
-            (Mode::Alphanumeric, Mode::Numeric) | (Mode::Byte, _) => Some(Ordering::Greater),
+            (Self::Numeric, Self::Alphanumeric) | (_, Self::Byte) => Some(Ordering::Less),
+            (Self::Alphanumeric, Self::Numeric) | (Self::Byte, _) => Some(Ordering::Greater),
             _ => None,
         }
     }
 }
 
 #[cfg(test)]
 mod mode_tests {
+    use std::cmp::Ordering;
+
     use crate::types::Mode::{Alphanumeric, Byte, Kanji, Numeric};
 
     #[test]
     fn test_mode_order() {
         assert!(Numeric < Alphanumeric);
         assert!(Byte > Kanji);
-        assert!(!(Numeric < Kanji));
-        assert!(!(Numeric >= Kanji));
+        assert!(matches!(Numeric.partial_cmp(&Kanji), None | Some(Ordering::Equal | Ordering::Greater)));
+        assert!(matches!(Numeric.partial_cmp(&Kanji), None | Some(Ordering::Less)));
```

Comment:
??????

Reply:
This is the result of applying <https://rust-lang.github.io/rust-clippy/master/index.html#neg_cmp_op_on_partial_ord>. I'll revert this.